packages/loader/container-loader/src/container.ts

```diff
@@ -1682,9 +1683,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
         // Both protocol and context should not be undefined if we got so far.
 
-        if (this._context?.disposed === false) {
-            this.context.setConnectionState(state, this.clientId);
-        }
+        this.setContextConnectedState(state, this._deltaManager.connectionManager.readOnlyInfo.readonly ?? false);
```

Comment:
I spent some time trying to understand ConnectionManager.readonly v. ConnectionManager.readonlyInfo.readonly, because I was concerned that using this one here but this.readonly above (that's what's passed to the "readonly" event) could cause inconsistencies.

They're identical except readonlyInfo also checks storageOnly mode.  _However_, from my reading of the code, storageOnly always implies readonly permissions (see implementation of NoDeltaStorage class and the `claims` it simulates). So I believe they will always be identical.

So... there might be some cleanup here, not sure if it's worth opening a follow-up item.  Take a look and see if you agree.

Reply:
To be clear - I think this change is fine.  It's just a little hard to convince myself all the corner cases are accounted for.  So maybe worth adding some tests?  Looks like connectionManager.spec.ts has no test coverage for this stuff.  Again, could be done in a follow-up.

Reply:
AB#1851